modules/oe_multilingual_url_suffix/src/Form/NegotiationUrlSuffixForm.php

```diff
@@ -0,0 +1,138 @@
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('language_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'language_negotiation_configure_url_suffix_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['language.negotiation'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    global $base_url;
+    $config = $this->config('language.negotiation');
+
+    $form['suffix'] = [
+      '#type' => 'details',
+      '#tree' => TRUE,
+      '#title' => $this->t('Path suffix configuration'),
+      '#open' => TRUE,
+      '#description' => $this->t('Language codes or other custom text to use as a path suffix for URL language detection. For the selected fallback language, this value may be left blank. <strong>Modifying this value may break existing URLs. Use with caution in a production environment.</strong> Example: Specifying "deutsch" as the path suffix code for German results in URLs like "example.com/contact_deutsch".'),
+    ];
+
+    $languages = $this->languageManager->getLanguages();
+    $suffixes = $config->get('url_suffixes');
+    foreach ($languages as $langcode => $language) {
+      $t_args = ['%language' => $language->getName(), '%langcode' => $language->getId()];
+      $form['suffix'][$langcode] = [
+        '#type' => 'textfield',
+        '#title' => $language->isDefault() ? $this->t('%language (%langcode) path suffix (Default language)', $t_args) : $this->t('%language (%langcode) path suffix', $t_args),
+        '#maxlength' => 64,
+        '#default_value' => isset($suffixes[$langcode]) ? $suffixes[$langcode] : substr($langcode, 0, 2),
+        '#field_prefix' => $base_url . '/index_',
+      ];
+    }
+
+    $form_state->setRedirect('language.negotiation');
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    $languages = $this->languageManager->getLanguages();
+
+    // Count repeated values for uniqueness check.
+    $count = array_count_values($form_state->getValue('suffix'));
+    foreach ($languages as $langcode => $language) {
+      $value = $form_state->getValue(['suffix', $langcode]);
+      if ($value === '') {
+        // Throw a form error if the suffix is blank for any language,
+        // although it is required for selected negotiation type.
+        $form_state->setErrorByName("suffix][$langcode", $this->t('The suffix may only be left blank for the <a href=":url">selected detection fallback language.</a>', [
+          ':url' => $this->getUrlGenerator()->generate('language.negotiation_selected'),
+        ]));
+      }
+      elseif (strpos($value, '/') !== FALSE) {
```

Comment:
This was relevant for the prefix but I'm wondering if it still applies for us. Potentially the language suffix could be "en/us" and this should work because the resulting URL should be http://example.com/contact_en/us
I think we should check for underscore, since that is what we use to retrieve the language code.